pkg/utils/file/download_test.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"testing"
+
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+func TestDownloadWithCacheAndExtract(t *testing.T) {
+	ctx := context.Background()
+	cacheDir := "/tmp/cache"
+	src := "http://example.com/file.tar.gz"
```

Comment:
Don't depend on an external source, start a test server instead.

Reply:
In test server where can I place the files?

Reply:
import "net/http/httptest" and use `httptest.NewServer()` to create a test server, and use `gzip` to compress the content as http response, and put the output in the tmp dir.